### PR TITLE
Fix the return type for useLazyQuery

### PR DESCRIPTION
**Pull request by eliot-stripe**
- Links to documentation: https://www.apollographql.com/docs/react/api/react/hooks/#function-signature-1
- Link to GitHub or NPM: https://github.com/apollographql/apollo-client
- Type of contribution: fix

### CODE REVIEW

**Comment by AndrewSouthpaw** [comment]
LGTM. @eliot-stripe please CC [some of the other maintainers of the libdef](https://github.com/flow-typed/flow-typed/blob/b6681a43c0a2cfd6a71403779b8b7b2ebcec5ab9/definitions/npm/%40apollo/react-hooks_v3.x.x/CONTRIBUTING.md) to get signoff from one of them.

**Comment by eliot-stripe** [comment]
@thymikee @rasmusprentow @bwlt any of you want to take a look?

**Review by bwlt — APPROVED** [review]

**Pull request merged**
